plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergUtil.java

```diff
@@ -35,20 +49,134 @@ public final class GlueIcebergUtil
+
+        if (cacheTableMetadata) {
+            // Store table metadata sufficient to answer information_schema.columns and system.metadata.table_comments queries, which are often queried in bulk by e.g. BI tools
+            String comment = metadata.properties().get(TABLE_COMMENT);
+            Optional<List<Column>> glueColumns = glueColumns(typeManager, metadata);
+
+            boolean canPersistComment = (comment == null || comment.length() <= GLUE_TABLE_PARAMETER_LENGTH_LIMIT);
+            boolean canPersistColumnInfo = glueColumns.isPresent();
+            boolean canPersistMetadata = canPersistComment && canPersistColumnInfo;
+
+            if (canPersistMetadata && false) {
+                tableInput.withStorageDescriptor(new StorageDescriptor()
+                        .withColumns(glueColumns.get()));
+
+                if (comment != null) {
+                    parameters.put(TABLE_COMMENT, comment);
+                }
+                else {
+                    parameters.remove(TABLE_COMMENT);
+                }
+                parameters.put(TRINO_TABLE_METADATA_INFO_VALID_FOR, newMetadataLocation);
+            }
+        }
+
+        tableInput.withParameters(parameters);
+
+        return tableInput;
+    }
+
+    private static Optional<List<Column>> glueColumns(TypeManager typeManager, TableMetadata metadata)
+    {
+        List<Types.NestedField> icebergColumns = metadata.schema().columns();
+        ImmutableList.Builder<Column> glueColumns = builderWithExpectedSize(icebergColumns.size());
+
+        boolean firstColumn = true;
+        for (Types.NestedField icebergColumn : icebergColumns) {
+            String glueTypeString = toGlueTypeStringLossy(icebergColumn.type());
+            if (glueTypeString.length() > GLUE_COLUMN_TYPE_LENGTH_LIMIT) {
+                return Optional.empty();
+            }
+            String trinoTypeName = TypeConverter.toTrinoType(icebergColumn.type(), typeManager).getTypeId().getId();
+            Column column = new Column()
+                    .withName(icebergColumn.name())
+                    .withType(glueTypeString)
+                    .withComment(icebergColumn.doc());
+            if (firstColumn || !glueTypeString.equals(trinoTypeName)) {
+                if (trinoTypeName.length() > GLUE_COLUMN_PARAMETER_LENGTH_LIMIT) {
+                    return Optional.empty();
+                }
+                // Store type parameter for some (first) column so that we can later detect whether column parameters weren't erased by something.
+                column.setParameters(ImmutableMap.of(COLUMN_TRINO_TYPE_PROPERTY, trinoTypeName));
+            }
+            glueColumns.add(column);
+
+            firstColumn = false;
+        }
+
+        return Optional.of(glueColumns.build());
+    }
+
+    // Copied from org.apache.iceberg.aws.glue.IcebergToGlueConverter#toTypeString
+    private static String toGlueTypeStringLossy(Type type)
```

Comment:
Did you start any discussion with the Iceberg community to move `IcebergToGlueConverter#toTypeString` to a utility class so that we reference it directly and not worry about going side-ways in the near to medium future?

Reply:
there already is a utility method `org.apache.iceberg.aws.glue.IcebergToGlueConverter#toTypeString`. i can use it directly, but didn't want to pull additional dependency just for that